x-pack/test/api_integration/apis/ml/calendars/helpers.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Calendar, CalendarEvent } from '../../../../../plugins/ml/server/models/calendar';
+
+export const allEventsExistInCalendar = (
+  eventsToCheck: CalendarEvent[],
+  calendar: Calendar
+): boolean => {
+  const updatedCalendarEvents = calendar.events as CalendarEvent[];
+  let allEventsAreUpdated = true;
+  for (const eventToCheck of eventsToCheck) {
+    // if at least one of the events that we need to check is not in the updated events
+    // no need to continue
+    if (
+      updatedCalendarEvents.findIndex(
+        (updatedEvent) =>
+          updatedEvent.description === eventToCheck.description &&
+          updatedEvent.start_time === eventToCheck.start_time &&
+          updatedEvent.end_time === eventToCheck.end_time
+      ) < 0
```

Comment:
While this logic is working, it seems a bit complicated and also doesn't give us a useful failure message.
I think something like this (partly pseudo code)
```
expect(updatedCalendarEvents).to.contain(eventToCheck, `Expected calendar X to have event JSON.stringify(eventToCheck)`)
```
would also work (and give better readability plus failure message).

Reply:
I did this way because the shape of the updatedEvent coming back includes extra fields like `calendar_id` and `event_id` so a strict check of object equality would be difficult.

Reply:
I have moved this utility function to api level with a detailed error message here https://github.com/elastic/kibana/pull/72564/commits/27bfb01ac70bf143daf8799d8d50d5a547df1cfc. This is how it would like like now if the check fails:
<img width="1791" alt="Screen Shot 2020-07-22 at 12 15 42 PM" src="https://user-images.githubusercontent.com/43350163/88208307-a2bf1500-cc16-11ea-824a-21972622e008.png">

Ideally we